Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,315 @@ public static class BalanceStats
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsStrArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            CultureInfo.CurrentCulture = GetCustomInvariantCulture();
+
+            decimal min = decimal.MaxValue;
+            string minResult = "";
+            string peopleWithSameLowBalance = "";
+            bool moreThanOne = false;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] auxBalances = peopleAndBalances[i].Split(',');
+                decimal[] balances = GetBalances(auxBalances[1..]);
+
+                if (balances.Length <= 1)
+                {
+                    return "N/A.";
+                }
+         
+                decimal currentMin = MinBalance(GetDiff(balances));
+                if (min > currentMin)
+                {
+                    min = currentMin;
+                    peopleWithSameLowBalance = auxBalances[0];
+                    moreThanOne = false;
+                    minResult = $" lost the most money. {min:C0}.";
+                }
+                else if (min.Equals(currentMin))
+                {
+                    peopleWithSameLowBalance += $", {auxBalances[0]}";
+                    moreThanOne = true;
+                }
+            }
+
+            if (moreThanOne)
+            {
+                peopleWithSameLowBalance = ReplaceLastCommaWithAnd(peopleWithSameLowBalance);
+            }
+            return peopleWithSameLowBalance + minResult;
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if(IsStrArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            CultureInfo.CurrentCulture = GetCustomInvariantCulture();
+
+            decimal max = decimal.MinValue;
+            string result = "";
+            string richestPeople = "";
+            bool moreThanOne = false;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] auxBalances = peopleAndBalances[i].Split(',');
+
+                string value = auxBalances[^1..][0];
+                decimal.TryParse(value, out decimal currentBalance);
+
+                if (max < currentBalance)
+                {
+                    max = currentBalance;
+                    richestPeople = auxBalances[0];
+                    moreThanOne = false;
+                    result = $" is the richest person. {max:C0}.";
+                }
+                else if(max.Equals(currentBalance))
+                {
+                    richestPeople += $", {auxBalances[0]}";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5069037)